package-template/.github/workflows/publish.yml

```diff
@@ -0,0 +1,32 @@
+name: Publish Package for the First Time
+
+on:
+  workflow_dispatch:
+
+env:
+  PPR_AUTH: ${{ secrets.PPR_AUTH }}
+
+jobs:
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+
+    steps:
+    # We need to have at least one tag available prior to publication. So we will publish the first release
+    # during publication.
+    - name: Checkout the Latest Package Code
+      uses: actions/checkout@v3
+    - name: Get Version
+      id: version
+      run: node -p "require('./package.json').version" >> "$GITHUB_OUTPUT"
+    - name: Create GitHub Release
+      uses: softprops/action-gh-release@v1
+      with:
+        tag_name: ${{ steps.version.outputs }}
+    # Now this is what's needed to actually publish the package to PPR
+    - name: Publish Package to PPR
+      if: ${{ secrets.PUBLISHED != 'true' }}
```

Comment:
would it be possible or desired to have this workflow call github's API to change the workflow's `PUBLISHED` var, or some other way of doing it automatically?

Reply:
This is a really good idea. I know there are endpoints that can modify secrets, so I'll look into utilizing those here

Reply:
hmm, theres "repository secrets" that we all know, but there is also something called "repository variables"? located here:

<img width="1189" alt="image" src="https://github.com/pulsar-cooperative/.github/assets/49562048/6fefd1c5-d988-49f1-a27c-397d10fd1f4d">

since its not a sensitive value, maybe it would be better (or at least make more sense) if it were put here? but I'm not familliar with repository variables, so I have no idea about the difference on usage (if there is any)

Reply:
Interesting. I've never heard of repository variables. I'll have to look into it.

Also while on the topic, I was going to see if we could have the `PPR_AUTH` token as an organization secret, to avoid having to set it on each repo.

Reply:
I think we can? but i'm not completely sure ><

Reply:
Now that `pulsar-ternjs` has been setup, I can confirm variables were the way to go, thanks a ton for the suggestion!